crates/net/network/src/transactions/config.rs

```diff
@@ -18,13 +18,41 @@ pub struct TransactionsManagerConfig {
     pub transaction_fetcher_config: TransactionFetcherConfig,
     /// Max number of seen transactions to store for each peer.
     pub max_transactions_seen_by_peer_history: u32,
+    /// How new pending transactions are propagated.
+    #[cfg_attr(feature = "serde", serde(default))]
+    pub propagation_mode: TransactionPropagationMode,
 }
 
 impl Default for TransactionsManagerConfig {
     fn default() -> Self {
         Self {
             transaction_fetcher_config: TransactionFetcherConfig::default(),
             max_transactions_seen_by_peer_history: DEFAULT_MAX_COUNT_TRANSACTIONS_SEEN_BY_PEER,
+            propagation_mode: TransactionPropagationMode::default(),
+        }
+    }
+}
+
+/// Determines how new pending transactions are propagated to other peers in full.
+#[derive(Debug, Clone, Default)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum TransactionPropagationMode {
+    /// Send full transactions to sqrt2 of current peers.
+    #[default]
+    Sqrt2,
```

Comment:
what's `2` about this? 
```suggestion
    Sqrt,
```